### DISCUSSION TITLE
How to find Main entity through relation in this case?

### DISCUSSION THREAD

**Discussion by ADrejta** [mikro-orm/mikro-orm#2843] [Q&A]
I'm using Mikroorm and seem to have some trouble to do this sort of query.

I have Users entity and Posts entity. Posts have a deleted_at field to indicate if they are soft deleted (which I query on).

I am trying to find all users that either have no posts associated with them or all of the posts associated with the user are soft deleted:
This fetches all the users that have no posts
```
const users = userEntity.find({posts: {$eq: null}})
```
 However i cannot seem to find a way to do booth the query above but also get the users that all of their posts are soft deleted (deleted_at is not null)

Anyone have any ideas? 
Thanks beforehand.

**Comment by kennethsato** [comment]
Hello @ADrejta .

I think I have the same issue here.

Do you also have a OneToMany relationship on user?
```
  @OneToMany(() => Post, post => post.comercial)
  posts? = new Collection<Post>(this);
```

When I query on the inverse side (ManyToOne), I get the relationship populated. But like this, apparently the Collection is not initialized and it does not filter

**Reply by ADrejta** [reply]
Yes, i have a OneToMany relationship with the user just like you pointed out here. Did that solve the problem for you?

**Reply by kennethsato** [reply]
No. Unfortunately I have the same problem as yours.

I was considering going the other way around.
Finding the deleted posts and then finding the respective users.
But I'm pretty sure there must be a easier way to do that.

**Reply by B4nan** [reply]
Start with imagining what query you want to fire to accomplish this. Sounds like you should use `$or` where you have one query for the soft deleted comments and one for the non existing comments. Something like this could work:

```ts
const users = userEntity.find({
  $or: [
    { posts: {$eq: null} },
    { posts: { deletedAt: { $ne: null } } },
  ]
})
```

I'll be happy to take a look if you can provide a test case as in https://github.com/mikro-orm/mikro-orm/tree/master/tests/issues so I can better understand what your actual setup is.

**Reply by kennethsato** [reply]
Thanks @B4nan.
It worked for me.
I'm not sure if @ADrejta had the same problem, but in my case:
I was trying to find the users with undeleted posts like this:

```
  public async find(): Promise<User[]> {
    const users = await this.ormRepository.find(
      {
        posts: { deleted_at: null },
      },
      {
        populate: true,
      },
    );
    return users;
  }
```

But what I actually had to do was to also filter users without posts, like this:
```
  public async find(): Promise<User[]> {
    const users = await this.ormRepository.find(
      {
        $and: [
          {
            posts: { $ne: null },
          },
          { posts: { deleted_at: null } },
        ],
      },
      {
        populate: true,
      },
    );
    return users;
  }
```

**Reply by ADrejta** [reply]
> Start with imagining what query you want to fire to accomplish this. Sounds like you should use `$or` where you have one query for the soft deleted comments and one for the non existing comments. Something like this could work:
> 
> ```ts
> const users = userEntity.find({
>   $or: [
>     { posts: {$eq: null} },
>     { posts: { deletedAt: { $ne: null } } },
>   ]
> })
> ```
> 
> I'll be happy to take a look if you can provide a test case as in https://github.com/mikro-orm/mikro-orm/tree/master/tests/issues so I can better understand what your actual setup is.

The thing about this query that doesn't generate the right result is that if a users has deleted and non deleted posts that users will be returned.
For example:

User 1 has Posts 1 (not soft deleted) and 2 (soft deleted)
User 2 has Posts 3 (not soft deleted) and post 4 (not soft deleted)
User 3 has Posts 5 (soft deleted) and 6 (soft deleted)
User 4 has no posts

With the query that is proposed the users that will be returned will be User 1 , User 4 and User 3 since User 1  has one soft deleted Post but actually has another post that is not soft deleted. The result I'm looking for is to  return in this case User 3 that has all posts (in this case 2 posts) that are soft deleted and User 4 that has no posts.
